pkg/certdb/certdb.go

```diff
@@ -0,0 +1,20 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package certdb
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// DB stores uplink public keys.
+type DB interface {
+	// SavePublicKey adds a new bandwidth agreement.
+	SavePublicKey(context.Context, storj.NodeID, crypto.PublicKey) error
+	// GetPublicKey gets the public key of uplink corresponding to uplink id
+	GetPublicKey(context.Context, storj.NodeID) (*ecdsa.PublicKey, error)
```

Comment:
This should also return crypto.PublicKey, we aren't only targeting ecdsa anymore.